src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [a, b, c];
+
+    if (sides.some((s) => s <= 0)) {
+      throw new Error('Invalid sides');
+    }
 
-export class Triangle implements Figure {}
+    const largest: number = Math.max(...sides);
 
-export class Circle implements Figure {}
+    sides.splice(sides.indexOf(Math.max(...sides)), 1);
+
+    if (largest >= sides.reduce((acc, cur) => acc + cur)) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - The error message should be more descriptive. Consider changing it to 'The provided sides do not form a valid triangle, as the sum of two sides must always be greater than the third side'.